src/components/picker/menu/index.tsx

```diff
@@ -1,111 +1,164 @@
-			this.addHandler()
+			this.valid = !Array.from(this.options.values()).find(option => option.value == this.search)
+			const search = this.search.toLocaleLowerCase()
+			for (const option of chain(this.options.values(), this.backend.values())) {
+				const value = option.value.toString().toLocaleLowerCase().includes(search)
+				const searches = option.search.some(value => value.includes(search))
+				const result = value || searches
+				option.set.visible(result)
+			}
 		}
 	}
 	addHandler() {
 		const validation = !this.validator ? true : this.validator(this.search)
 		if (typeof validation == "object" ? validation.result : validation) {
 			if (!this.multiple)
-				for (const option of this.options.values())
-					option.element.selected = false
-			this.new = [...this.new, { value: this.search, selected: true }]
+				for (const option of chain(this.options.values(), this.backend.values()))
+					option.set.selected(false)
+			this.created = new Map(this.created.set(this.search, { value: this.search, selected: true }).entries())
 			this.search = ""
-			this.searchElement?.focus()
+			this.valid = false
 		}
 		if (typeof validation == "object")
 			this.notice.emit(validation.notice)
 	}
+	keyDownHandler(event: KeyboardEvent) {
+		if (event.key == "Enter") {
+			event.preventDefault()
+			this.addHandler()
+		}
+	}
+	@Listen("smoothlySlotEmpty")
+	emptyDisplayHandler(event: CustomEvent<Slot>) {
+		event.stopPropagation()
+		event.detail.set.nodes(this.display)
+	}
 	render() {
 		return (
-			<Host>
+			<Host class={{ valid: this.valid }}>
+				<smoothly-slotted-elements class={"hide"} onSmoothlySlottedChange={e => (this.display = e.detail)}>
+					<slot name="display" />
+				</smoothly-slotted-elements>
+				<div class={"hide"}>
+					<slot />
+					{Array.from(this.created.values(), option => (
+						<smoothly-picker-option key={option.value} value={option.value} selected={option.selected}>
+							{option.value}
+							<smoothly-slot-elements slot="display" nodes={this.display} />
+						</smoothly-picker-option>
+					))}
+				</div>
 				<div class={"controls"}>
 					<smoothly-input
-						ref={element => (this.searchElement = element)}
 						name="search"
 						value={this.search}
 						onSmoothlyInput={e => this.inputHandler(e)}
-						onSmoothlyChange={e => this.inputHandler(e)}
-						onSmoothlyBlur={e => e.stopPropagation()}
-						onKeyDown={event => this.keyDownHandler(event)}>
+						onSmoothlyChange={e => e.stopPropagation()}
+						onSmoothlyBlur={e => e.stopPropagation()}>
 						<slot name="search" />
 					</smoothly-input>
-					{this.mutable ? (
-						<button onClick={() => this.addHandler()} disabled={!this.allowed} class={"add"} type={"button"}>
+					{this.mutable && !this.readonly ? (
+						<button onClick={() => this.addHandler()} disabled={!this.valid} type={"button"}>
 							<smoothly-icon name="add-outline" />
 						</button>
-					) : null}
+					) : (
+						<Fragment></Fragment>
```

Comment:
Did we agree on this? I prefer null to indicate that nothing more happens. I can understand wrapping with Fragment instead of array, but here I think null is more straightforward.

Reply:
moved back to `null` but `<Fragment>` is "new" and fuuuuuun :smile: